src/App.js

```diff
@@ -36,51 +39,32 @@ class App extends Component {
           mainGenres: genres.genres1811
         })
       })
-      .then(() => this.getFilters())
       .catch(error => {
         throw new Error(error)
       })
   }
-  getFilters = () => {
+  getFilters = (key) => {
     let filters = [];
-    let filtersTwo = [];
     this.state.games.forEach(game => {
-      game.platforms.forEach(val => {
-        console.log("in filter1")
+      game[key].forEach(val => {
         if (!filters.includes(val)){
           filters.push(val)
         }
       })
     })
-    this.state.games.forEach(game => {
-      game.genres.forEach(val => {
-        if (!filtersTwo.includes(val)){
-          filtersTwo.push(val)
-        }
-      })
-    })
-    this.setState({platforms: filters})
-    this.setState({genres: filtersTwo})
-                  
+    this.setState({[key]: filters})
   }
   updateState = (newState) => {
     this.setState(newState);
   }
- 
   render() {
-    console.log("games", this.state.games);
-    console.log("plat", this.state.platforms);
-    console.log("genre", this.state.genres);
     return (
       <div className="app">
         <Header />
-        <GameShelf 
-          games={this.state.games}/>
+        <GameShelf {...this.state}/>
         <Footer {...this.state}
```

Comment:
Nice use of the spread operator, just make sure that if you're passing in the *entire* state object, that the component your passing them into actually needs every piece of information in that state object. If it doesn't, you will be triggering unnecessary re-renders to that component no matter what information in state changes, and you lose some of the performance benefits of react.